Source/WebGPU/WebGPU/XRProjectionLayer.mm

```diff
@@ -51,7 +54,55 @@
 
 bool XRProjectionLayer::isValid() const
 {
-    return true;
+    return !!m_colorTextures;
+}
+
+id<MTLTexture> XRProjectionLayer::colorTexture()
+{
+    return m_colorTexture;
+}
+
+id<MTLTexture> XRProjectionLayer::depthTexture()
+{
+    return m_depthTexture;
+}
+
+id<MTLSharedEvent> XRProjectionLayer::completionEvent()
+{
+    return m_sharedEvent;
+}
+
+void XRProjectionLayer::startFrame(size_t frameIndex, WTF::MachSendRight&& colorBuffer, WTF::MachSendRight&& depthBuffer, WTF::MachSendRight&& completionSyncEvent, size_t reusableTextureIndex)
+{
+#if !PLATFORM(IOS_FAMILY_SIMULATOR)
+    id<MTLDevice> device = m_device->device();
+    NSNumber* textureKey = @(reusableTextureIndex);
+    if (colorBuffer.sendRight()) {
+        MTLSharedTextureHandle* m_colorHandle = [[MTLSharedTextureHandle alloc] initWithMachPort:colorBuffer.sendRight()];
```

Comment:
Should this be `auto colorHandle = adoptNS([[MTLSharedTextureHandle alloc] initWithMachPort:colorBuffer.sendRight()]);`?

Reply:
WebGPU uses ARC so it would be fine but is not needed